File: delete.cpp
#include <iostream>
#include <stack>
#include <queue>
#include<string>

using namespace std;

int newArr[500];

int* func(int arr[], int size){
	queue<int> q;
	q.push(arr[0]);

	int p = 0;
	while(queue.empty()){
		int i=p+1;
		while(i<size){
			if (arr[i] > arr[p]){
				q.pop();
				newArr[p] = arr[p];
				p++;
			}
		}
		q.push(arr[i]);
	}
	newArr[p] = -1;
	return &newArr[0];
}

int main(){
	int size = 10;
	int arr[] {4,2,3,8,9,0};
	int *p;
	p = func(arr, 10);
	for (int j=0; j<size; j++)
		cout<<*(p+j);
	return 0;
}



//decemer cookoff challenge-beginner level - change it

#include<bits/stdc++.h>
#define MOD 1000000007
#define ll long long int
#define debug(x) cout<<"x :"<<x<<endl;
#define case(t,ans) cout<<"Case #"<<t<<": "<<ans<<endl;
using namespace std;

void solve();

int main(){
    #ifndef ONLINE_JUDGE
    freopen("input.txt", "r", stdin);
    freopen("output.txt", "w", stdout);
    #endif
    ios_base::sync_with_stdio(false);
    cin.tie(NULL);
    
    solve();
    
    
    return 0;
}

void solve()
{
    int t;
    cin>>t;
    
    while(t--)
    {
        int n;
        cin>>n;
        
        int arr[n];
        int store[100];
       memset(store,0,sizeof(store));
        for(int i=0;i<n;i++)
        {
            cin>>arr[i];
            store[arr[i]-1]++;
        }
        
        int max=0;
        for(int i=0;i<100;i++)
        {
            if(store[i]>max)
            {
                max=store[i];
            }
        }
        
        cout<<n-max<<endl;
     
    }
}

File: competitive_programming-II/delete.py
import sys 
sys.stdin = open('input.txt', 'r') 
sys.stdout = open('output.txt', 'w')
def solve(ans,l,d,n,x):
    re=0
    while(ans>0):
        start=ans
        for i in range(0,n):
            if(start>d):
                a[ans]=0
                re=0
                ans-=1
                # print("start: ",start)
                break
            if(start%l[i]==0):
                    re=1
            else:
                # print("start: ",start)
                r=l[i]-(start%l[i])
                rr=(start%l[i])
                start+=r
                if(start>d):
                    re=0
                    # print("start: ",start)
                    # if(ans%l[i]!=0):
                    #     a=ans//l[i]
                    #     if(a>0):
                    #         ans=(a)*l[i]
                    # else:
                    ans-=1
                    break
        if(re):
            break
    print("Case #",end="")
    print(x,end="")
    print(": ",end="")
    print(ans)
    return

t=int(input())
x=0
while(t):
    t-=1
    x+=1
    n,d=map(int,input().split())
    l=list(map(int,input().split()))
    ans=d
    lis=[]
    lis.append(l[0])
    for i in range(1,n):
        if(l[i-1]%l[i]==0):
            pass
        else:
            lis.append(l[i])
    n=len(lis)
    # l.sort(reverse=True)
    solve(ans,lis,d,n,x)